bundles/org.openhab.core.semantics/model/generateTagClasses.groovy

```diff
@@ -69,6 +69,8 @@ def createTagSetClass(def line, String tagSet) {
     def parent = line.Parent
     def parentClass = parent ? parent : type
     def pkg = type.toLowerCase()
+    def ch = label.toLowerCase().charAt(0)
+    def article = ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' ? "an" : "a"
```

Comment:
Given that the correct choice depends on how the word is pronounced, I'm not certain this rule, if I'm reading it correctly, will work all the time. For example, it's "an herb" but "a history". There are already lots of examples in the changes above where most words starting with "u" use "a" and not "an". 

But for these purposes it might be close enough. I'm not sure the purpose of these in relation to Groovy and such.

Reply:
If I remember English (UK) grammer correct, there is a clear rule to use „an“ only followed by vowels in writing.